Marlin/MarlinSerial.h

```diff
@@ -75,27 +75,27 @@
 #define BIN 2
 #define BYTE 0
 
-#ifndef USBCON
-  // Define constants and variables for buffering incoming serial data.  We're
-  // using a ring buffer (I think), in which rx_buffer_head is the index of the
-  // location to which to write the next incoming character and rx_buffer_tail
-  // is the index of the location from which to read.
-  // Use only powers of 2.
-  // : [0, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, ...]
-  #ifndef RX_BUFFER_SIZE
-    #define RX_BUFFER_SIZE 128
-  #endif
-  // 256 is the max TX buffer climit due to uint8_t head and tail.
-  #ifndef TX_BUFFER_SIZE
-    #define TX_BUFFER_SIZE 32
-  #endif
+// Define constants and variables for buffering serial data.
+// Use only 0 or powers of 2 greater than 1
+// : [0, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, ...]
+#ifndef RX_BUFFER_SIZE
+  #define RX_BUFFER_SIZE 128
+#endif
+// 256 is the max TX buffer limit due to uint8_t head and tail.
+#ifndef TX_BUFFER_SIZE
+  #define TX_BUFFER_SIZE 32
+#endif
 
+#ifndef USBCON
```

Comment:
My biggest question is why all these tests can be run even if USBCON is defined ?  But it's 1.1.x so I'm not going to loose any sleep over it ;)

Reply:
It looks like we're assuming those options can't be overridden on platforms that have `USBCON`. I'd have to reach back into the Git Blame annals to find out the reasoning.